File: server/product-service/src/main/kotlin/com/sudoo/productservice/model/ProductInfo.kt
package com.sudoo.productservice.model

import org.springframework.data.annotation.Transient
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.time.LocalDateTime

@Table("products")
data class ProductInfo(
    @Column("product_id")
    val productId: String,
    @Column("supplier_id")
    val supplierId: String,
    @Column("sku")
    val sku: String,
    @Column("name")
    val name: String,
    @Column("brand")
    var brand: String,
    @Column("price")
    val price: Float,
    @Column("listed_price")
    val listedPrice: Float,
    @Column("amount")
    val amount: Int,
    @Column("saleable")
    val saleable: Boolean,
    @Column("rate")
    val rate: Float,
    @Column("discount")
    val discount: Int,
    @Column("start_date_discount")
    val startDateDiscount: LocalDateTime?,
    @Column("end_date_discount")
    val endDateDiscount: LocalDateTime?,
) {

    @Transient
    var images: List<String>? = null
}

File: client/web/sudoo/lib/domain/model/discovery/product_info.dart
class ProductInfo {
  final String productId;
  final String sku;
  final String name;
  final double price;
  final double listedPrice;
  final int amount;
  final double rate;
  final int discount;
  final DateTime? startDateDiscount;
  final DateTime? endDateDiscount;
  final bool saleable;
  final String brand;
  final List<String> images;

  const ProductInfo(
    this.productId,
    this.sku,
    this.name,
    this.price,
    this.listedPrice,
    this.amount,
    this.rate,
    this.discount,
    this.startDateDiscount,
    this.endDateDiscount,
    this.saleable,
    this.brand,
    this.images,
  );
}
